tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿namespace Sorry.Analyzers.Tests
+{
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using RoslynTestKit;
+    using Xunit;
+
+    public class RR1337IfStatementInTestMethodTests : AnalyzerTestFixture
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        [Fact]
+        public void XunitFactAttributeTest()
+        {
+            const string text = @"
+namespace Tests;
+
+using Xunit;
+
+public class Foo
+{
+    [Fact]
+    public void MyTestMethod()
+    {
+        [|if (true)|]
+        {
+        }
+    }
+}
+";
+
+            this.HasDiagnostic(
+                markupCode: text,
+                diagnosticId: DiagnosticIds.Sorry1337);
+        }
+
+        [Fact]
+        public void XunitTheoryAttributeTest()
+        {
+            const string text = @"
+namespace Tests;
+
+using Xunit;
+
+public class Foo
+{
+    [Theory]
+    public void MyTestMethod()
+    {
+        [|if (true)|]
+        {
+        }
+    }
+}
+";
+
+            this.HasDiagnostic(
+                markupCode: text,
+                diagnosticId: DiagnosticIds.Sorry1337);
+        }
+
+        [Fact]
+        public void NunitTestAttributeTest()
+        {
+            const string text = @"
+namespace Tests;
+
+using NUnit.Framework;
+
+public class Foo
+{
+    [Test]
+    public void MyTestMethod()
+    {
+        [|if (true)|]
+        {
+        }
+    }
+}
+";
+
+            this.HasDiagnostic(
+                markupCode: text,
+                diagnosticId: DiagnosticIds.Sorry1337);
+        }
+
+        [Fact]
+        public void NunitTestCaseAttributeTest()
```

Comment:
May I kindly ask you to add some tests on negative scenarios? For example, DoesNotWarnIfInsideRegularMethod

Reply:
Thank you for your suggestion. I added some negative test cases.

Reply:
Great job. But the case when if is located inside the regular method (without any attributes) is still missing.

Reply:
Thanks. I added test case to cover it. And another one to cover non-test attribute.